### PR TITLE
numpydoc style (signac/contrib/import_export.py)

### PR DESCRIPTION
**Pull request by ac-optimus**
## Description

issue #315 
## Types of Changes

- [x] Documentation update
- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac/blob/master/CONTRIBUTING.md#code-style) of this project.
- [x] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [ ] I have updated the [changelog](https://github.com/glotzerlab/signac/blob/master/changelog.txt) and added all related issue and pull request numbers for future reference (if applicable). See example below.

Change of doc string to `numpydoc` in import_export.py

### CODE REVIEW

**Comment by mikemhenry** [comment]
Just giving this a PR a bump, @ac-optimus let me know if you have any questions or if there are any blockers.

**Comment by bdice** [comment]
@ac-optimus Let us know if you'd like someone else to finish up this PR or if you'd like to finish it (it's still marked as a draft). Thanks! 👍

**Comment by ac-optimus** [comment]
> @ac-optimus Let us know if you'd like someone else to finish up this PR or if you'd like to finish it (it's still marked as a draft). Thanks!

hi, sure I would be happy if someone else could take this up.

**Comment by bdice** [comment]
@ac-optimus Thanks for your help. I finished up this PR. @csadorf and @vishav1771, please review when convenient. 👍

File: signac/contrib/import_export.py [hunk 1]
```diff
@@ -91,20 +147,52 @@ def format_field(self, value, format_spec):
...
     if path is None:
         # Generate a path function based on the schema detected for jobs.
         path_function = _make_schema_based_path_function(jobs=jobs)
 
     elif path is False:
 
-        # Just use the job-id as path.
+        # Just use the job id as path.
         def path_function(job):
+            """Create path function when path is not given.
```
Comment (csadorf) [inline signac/contrib/import_export.py:183]: I'm not sure that we really need to document these internal closures, but if we do then this doc-string is not correct.
```suggestion
            """Use job-id to construct path.
```

File: signac/contrib/import_export.py [hunk 2]
```diff
@@ -117,6 +205,19 @@ def path_function(job):
         paths = _make_schema_based_path_function(jobs=jobs, exclude_keys=exclude_keys)
 
         def path_function(job):
+            """Format a path for a given job.
```
Comment (csadorf) [inline signac/contrib/import_export.py:208]:
```suggestion
            """Format a path based on a given string and schema.
```
There might be a better way to describe this, but the current doc-string is very generic. If we go with that generic description, then we should keep it generic for the other path functions as well.

File: signac/contrib/import_export.py [hunk 3]
```diff
@@ -145,7 +246,19 @@ def path_function(job):
 
 
 def _check_directory_structure_validity(paths):
-    "Check the consistency of the directory structure for export."
+    """Validate directory structure.
```
Comment (csadorf) [inline signac/contrib/import_export.py:249]:
```suggestion
    """Validate directory structure consistency.
```

File: signac/contrib/import_export.py (hunk 1 above)
Comment (bdice) [reply signac/contrib/import_export.py:183]: @csadorf `pydocstyle` does require us to document internal closures, anything with a `def`. Thanks for the suggestion.

**Pull request merged**